paramiko/sftp_client.py

```diff
@@ -136,6 +136,7 @@ def __init__(self, sock):
                 server_version
             ),
         )
+        self._cwd = b(self.normalize("."))
```

Comment:
In order to call self.normalize before self._cwd is defined, I'd have to change self._adjust_cwd to check hasattr before reading self._cwd. Want me to do that?

Reply:
Aha, there was a reason for it being placed lower down. Nah, leave it as-is, bitprophet can evaluate.

Reply:
Although: based on his ❤️ of my https://github.com/paramiko/paramiko/pull/2224#issuecomment-1545768553, I would say that any implementation here _does_ need to provide access to the original behavior.

And, since it's going to be a toggle flag either way, it's going to be less trouble if we retain the default of non-initialization and add this `_cwd` init as an option.

And actually, now that I look more closely -- if the `_cwd` initialization is an opt-in, it seems like it actually _wouldn't_ be that much more complicated to have that opt-in argument be a string, for the path to use for `_cwd` initialization; something like:

```
def __init__(self, sock, *, cwd_init_path=None):
    ...
    if cwd_init_path:
        self._cwd = b(self.normalize(cwd_init_path))
```

(Seems to me that an empty string for this initialization would be a bad idea, and thus my proposing the `if cwd_init_path:`)